### PR TITLE
[#4153] Use COinS to provide Zotero integration on search results page, rather than UnAPI

### LINKED ISSUE #4153: Rails 7 upgrade triggers an error when hitting https://catalog.princeton.edu/unapi
### Description of the issue
After upgrading to blacklight 7.38.0 and Rails 7 we're seeing [honeybadger errors](https://app.honeybadger.io/projects/54399/faults/110773459) related to 'unapi'.

### Actual behavior
This link `<link rel="unapi-server" type="application/xml" title="unAPI" href="[/unapi](https://catalog.princeton.edu/unapi)"/>` is loaded in the [application.html](https://github.com/pulibrary/orangelight/blob/main/app/views/layouts/application.html.erb#L45).

### Notes
Dropdown 'Send' -> 'RIS' exports the ris file and imports it in the Zotero app.
The [invalid parameter handler](https://github.com/pulibrary/orangelight/blob/main/lib/orangelight/middleware/invalid_parameter_handler.rb) was changed to send the bad request to the load balancer and not to a custom error page

### Honeybadger link and code snippet, if applicable
[Honeybadger error 110773459](https://app.honeybadger.io/projects/54399/faults/110773459)

```
ActionView::MissingTemplate: Missing template catalog/formats.xml with {:locale=>[:en], :formats=>[:html, :text, :js, :css, :ics, :csv, :vcf, :vtt, :png, :jpeg, :gif, :bmp, :tiff, :svg, :mpeg, :mp3, :ogg, :m4a, :webm, :mp4, :otf, :ttf, :woff, :woff2, :xml, :rss, :atom, :yaml, :multipart_form, :url_encoded_form, :json, :pdf, :zip, :gzip, :jsonld, :refworks_marc_txt, :openurl_kev, :endnote, :marc, :marcxml, :openurl_ctx_kev, :dc_xml, :oai_dc_xml, :rdf_dc, :ris, :mendeley], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :jbuilder]}.

Searched in:
  * "/opt/orangelight/releases/20240828230910/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/health-monitor-rails-12.3.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/flipflop-2.7.1/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/devise-4.9.4/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/bundler/gems/blacklight_unapi-79ac4bf1914f/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/blacklight_range_limit-8.5.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/blacklight-marc-8.1.4/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/blacklight_dynamic_sitemap-1.0.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/blacklight_advanced_search-7.0.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/blacklight-7.38.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/view_component-2.83.0/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/kaminari-core-1.2.2/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/actiontext-7.0.8.4/app/views"
  * "/opt/orangelight/shared/bundle/ruby/3.1.0/gems/actionmailbox-7.0.8.4/app/views"

Did you mean?  catalog/formats
               catalog/facet
               catalog/sms
               catalog/email
               catalog/email_success
               catalog/endnote
```

### PR DESCRIPTION
**Pull request by sandbergja**
The Zotero documentation (https://www.zotero.org/support/dev/exposing_metadata#using_an_open_standard_for_exposing_metadata) lists three ways to expose metadata to zotero:
* `<meta>` tags (this doesn't work for a search results page, since it can only describe a single resource)
* COinS
* unAPI

Previously, we used unAPI on both the search results and show pages, and additionally provided COinS on the show page.  However, the blacklight-unapi gem has not been updated in 10 years and is not compatible with Rails 7.

This commit:
* removes unAPI from both search results and show pages
* adds COinS to the search results page

closes #4153

### CODE REVIEW

**Review by kevinreiss — APPROVED** [review]

File: app/views/catalog/_index_default.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <% doc_presenter = document_presenter(document) %>
 <%# default partial to display solr document fields in catalog index view -%>
-<abbr class="unapi-id" title="<%= document.id %>" aria-hidden="true"></abbr>
+<span class="Z3988" aria-hidden='true' title="<%= document.export_as_openurl_ctx_kev %>"></span>
```
Comment (christinach) [inline app/views/catalog/_index_default.html.erb:3]: what is class Z3988 ? is it a random string?

**Review by christinach — APPROVED** [review]
Thanks @sandbergja ! one question.

File: app/views/catalog/_index_default.html.erb (hunk 1 above)
Comment (sandbergja) [reply app/views/catalog/_index_default.html.erb:3]: The Z3988 class is [part of the COinS standard](https://web.archive.org/web/20170424223448/http://ocoins.info/#id3205609419).  Apparently it is the NISO number for the OpenURL standard?

Comment (kevinreiss) [reply app/views/catalog/_index_default.html.erb:3]: It's part of the syntax of the COINS specification. https://en.wikipedia.org/wiki/COinS. I think it's an ISO code.

**Pull request merged**